### DISCUSSION TITLE
How to use a fixed port for Kafka external listener of type NodePort

### DISCUSSION THREAD

**Discussion by milon619** [strimzi/strimzi-kafka-operator#4395] [Q&A] [answered]
We want to allow external access to Strimzi Kafka cluster for some clients. Right now we are planning to use the external listener of type NodePort for accomplishing this. During testing, we found that the 'xxx-external-bootstrap' service is using a random port to listen on all Nodes where the Kafka broker is running. We are worried that if we recreate the OCP cluster where the Kafka brokers are running whether we will get the same port numbers? We are trying to avoid a situation where we have to reconfigure all Kafka external clients' configuration if the Openshift cluster is recreated as part of disaster recovery or something like that.

**Comment by scholzj** [comment, marked as answer]
You can use this to specify the node port which Strimzi should request for the services: https://strimzi.io/docs/operators/latest/full/using.html#property-listener-config-nodeport-reference

However, keep in mind that in general, the ports are assigned randomly by Kubernetes for the NodePort services. So you need to make sure that the ports you specify there will be available. If they are already used by some other application, you would need to move the application to another port first.

**Reply by milon619** [reply]
Thanks @scholzj that explains it well.
